tests/unit/test_client.py

```diff
@@ -138,6 +140,80 @@ async def test__get_metadata_with_psc(
     }
 
 
+async def test__get_metadata_error(
+    credentials: FakeCredentials,
+) -> None:
+    """
+    Test that AlloyDB API error messages are raised for _get_metadata.
+    """
+    # mock AlloyDB API calls with exceptions
+    client = AlloyDBClient(
+        alloydb_api_endpoint="https://alloydb.googleapis.com",
+        quota_project=None,
+        credentials=credentials,
+    )
+    get_url = "https://alloydb.googleapis.com/v1beta/projects/my-project/locations/my-region/clusters/my-cluster/instances/my-instance/connectionInfo"
+    resp_body = {
+        "error": {
+            "code": 403,
+            "message": "AlloyDB API has not been used in project 123456789 before or it is disabled",
+        }
+    }
+    with aioresponses() as mocked:
+        mocked.get(
+            get_url,
+            status=403,
+            payload=resp_body,
+            repeat=True,
+        )
+        with pytest.raises(ClientResponseError) as exc_info:
+            await client._get_metadata(
+                "my-project", "my-region", "my-cluster", "my-instance"
+            )
+        exc = exc_info.value
+        assert exc.status == 403
+        assert (
+            exc.message
+            == "AlloyDB API has not been used in project 123456789 before or it is disabled"
+        )
+    await client.close()
+
+
+async def test__get_metadata_error_parsing_json(
```

Comment:
what is the value of this test? I am not sure what exactly we are testing here...

Reply:
It covers the case when `await resp.json()` throws an exception.

Reply:
In that case I like this test but let's update the docstrings as really this test is not testing the AlloyDB API error messages at all, it should be testing that if the AlloyDB API's return a bad JSON then the aiohttp default responses are raised.

So instead of testing that the error message does not equal an arbitrary value like:
```python
assert exc_info.value.message != "my arbitrary message"
```

We should test that we are falling back to the default aiohttp error message. (403 -> Forbidden, 404 -> Not Found)

```
assert exc_info.value.message != "Forbidden"
```

Reply:
Done